azurerm_postgresql_active_directory_administrator: "string is required" when upgrading from AzureRM 3.14 to 3.15

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.2.5

### AzureRM Provider Version

3.15

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_active_directory_administrator

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.15.0"
    }
  }
}

provider "azurerm" {
  features {}
}

data "azurerm_client_config" "current" {}

resource "azurerm_resource_group" "example" {
  name     = "azurerm-debug"
  location = "canadaeast"
}

resource "azurerm_postgresql_server" "example" {
  name                         = "azurerm-debug-psqlserver"
  resource_group_name          = azurerm_resource_group.example.name
  location                     = azurerm_resource_group.example.location
  version                      = "11"
  administrator_login          = "administratorexample"
  administrator_login_password = "thisisaterriblepassworddontuseit12345!"
  sku_name                     = "GP_Gen5_2"
  ssl_enforcement_enabled      = true
}

resource "azurerm_postgresql_active_directory_administrator" "example" {
  server_name         = azurerm_postgresql_server.example.name
  resource_group_name = azurerm_resource_group.example.name

  login               = "sqladmin"
  tenant_id           = data.azurerm_client_config.current.tenant_id
  object_id           = data.azurerm_client_config.current.object_id
}
```


### Debug Output/Panic Output

```shell
https://gist.github.com/9numbernine9/bd59bd2fb995cb5d8c1f85f01dee16b6
```


### Expected Behaviour

Terraform will `plan` and/or `apply` successfully.

### Actual Behaviour

The following error is generated:
```
╷
│ Error: string is required
│ 
│   with azurerm_postgresql_active_directory_administrator.example,
│   on main.tf line 32, in resource "azurerm_postgresql_active_directory_administrator" "example":
│   32: resource "azurerm_postgresql_active_directory_administrator" "example" {
│ 
╵
```

### Steps to Reproduce

1. Change the Terraform configuration in this issue to use AzureRM `3.14`.
2. `terraform init -upgrade` 
3. `terraform apply` - this should work fine.
4. Switch to AzureRM `3.15`
5. `terraform init -upgrade`
6. `terraform plan` - Error message received.

### Important Factoids

This only occurs when upgrading a resource that was previously created with AzureRM `3.14` (possibly earlier versions as well). This error will _not_ occur if the resource is created from scratch using AzureRM `3.15`.

### References

_No response_

Have same problem. Snip from Trace:    

```
2022-07-22T17:03:27.914+0200 [ERROR] plugin.terraform-provider-azurerm_v3.15.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR diagnostic_summary="string is required" tf_provider_addr=provider @caller=github.com/hashicorp/terraform-plugin-go@v0.10.0/tfprotov5/internal/diag/diagnostics.go:56 tf_resource_type=azurerm_postgresql_active_directory_administrator diagnostic_attribute=AttributeName("id") tf_proto_version=5.2 tf_req_id=XXX-XXXX-XXXX-XXXXX tf_rpc=UpgradeResourceState @module=sdk.proto diagnostic_detail= timestamp=2022-07-22T17:03:27.914+0200
```

--- a/internal/services/postgres/migration/postgresql_aad_administrator.go
+++ b/internal/services/postgres/migration/postgresql_aad_administrator.go
@@ -62,7 +62,7 @@ func (PostgresqlAADAdministratorV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFun
 
 		newId := serveradministrators.NewServerID(id.SubscriptionId, id.ResourceGroup, id.ServerName)
 		log.Printf("[DEBUG] Updating ID from %q to %q", oldId, newId)
-		rawState["id"] = newId
+		rawState["id"] = newId.ID()
 
 		return rawState, nil
 	}

